File: environment-spring-boot/casadocodigo/src/main/java/org/ecorp/casadocodigo/services/AutorService.java
package org.ecorp.casadocodigo.services;

import java.util.Optional;
import javax.validation.Valid;
import org.ecorp.casadocodigo.dtos.AutorDTO;
import org.ecorp.casadocodigo.exceptions.ResourceNotFoundException;
import org.ecorp.casadocodigo.forms.AutorFormRequest;
import org.ecorp.casadocodigo.model.Autor;
import org.ecorp.casadocodigo.repositories.AutorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AutorService {

  @Autowired
  private AutorRepository repository;

  public AutorDTO create(@Valid AutorFormRequest autorForm) {

    Autor saved = repository.save(autorForm.map());
    return new AutorDTO(saved);

  }

  public AutorDTO buscaAutorById(Long id) {

    Optional<Autor> found = repository.findById(id);
    if (found.isEmpty()) {
      throw new ResourceNotFoundException();
    }
    return new AutorDTO(found.get());
  }



}

File: src/main/kotlin/com/tiozao/cdd/loja/domain/service/AutorService.kt
package com.tiozao.cdd.loja.domain.service

import com.tiozao.cdd.loja.domain.model.AutorModel
import com.tiozao.cdd.loja.repository.AutorRepository
import com.tiozao.cdd.loja.repository.extensions.toEntity
import com.tiozao.cdd.loja.repository.extensions.toModel
import org.springframework.stereotype.Service
import org.springframework.validation.annotation.Validated
import javax.validation.Valid


@Service
@Validated
class AutorService(private var autorRepository: AutorRepository) {

    fun createAutor(@Valid autor: AutorModel): AutorModel {
        return autorRepository.save(
            autor.toEntity())
            .toModel()
    }

    fun findAutor(autorId: Int): AutorModel {
        autorRepository.findById(autorId)?.let { return it.get().toModel() }
    }


}